### PR TITLE
Scripts/Spells: Port druid scripts to new script registration syntax (2 of 3)

### PR DESCRIPTION
**Pull request by Sorikoff**
**Changes proposed:**
Port almost all (some scripts left untouched) of `spell_druid.cpp` scripts to new script registration syntax. _No changes in functionality._

**Target branch(es):** 3.3.5

~~**Issues addressed:**~~

**Tests performed:** It does build. I tested some random spells and items affected by these changes. Seems to be working.

~~**Known issues and TODO list:**~~

### CODE REVIEW

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -507,7 +507,7 @@ class spell_dru_glyph_of_innervate : public AuraScript
         SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_DRUID_GLYPH_OF_INNERVATE_REGEN);
         int32 amount = CalculatePct(static_cast<int32>(caster->GetCreatePowerValue(POWER_MANA)), aurEff->GetAmount());
 
-        ASSERT(spellInfo->GetMaxTicks() > 0);
+                ASSERT(spellInfo->GetMaxTicks() > 0);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp]: don't break existing codestyle

**Review by Shauren — CHANGES_REQUESTED** [review]
I'm going to have to request you revert all codestyle changes (except reducing indentation that is a result of removing outer classes)

**Comment by Sorikoff** [comment]
> I'm going to have to request you revert all codestyle changes (except reducing indentation that is a result of removing outer classes)

Reverted parts outside of the scope of this PR.

**Comment by ForesterDev** [comment]
I think, you should convert complex Spell/Aura scripts too, using RegisterSpellAndAuraScriptPair

File: src/server/scripts/Spells/spell_druid.cpp [hunk 2]
```diff
@@ -996,1134 +996,916 @@ class spell_dru_omen_of_clarity : public AuraScript
...
-        {
-            PrepareAuraScript(spell_dru_revitalize_AuraScript);
-
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                return ValidateSpellInfo(
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
                 {
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp]: What are you using to do these mass changes? It looks like you don't properly handle all cases, all these lines should also have gotten -8 spaces

File: src/server/scripts/Spells/spell_druid.cpp [hunk 3]
```diff
@@ -996,1134 +996,916 @@ class spell_dru_omen_of_clarity : public AuraScript
...
-        {
-            PrepareAuraScript(spell_dru_t3_2p_bonus_AuraScript);
-
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                return ValidateSpellInfo(
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
                 {
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp]: same here

File: src/server/scripts/Spells/spell_druid.cpp [hunk 4]
```diff
@@ -996,1134 +996,916 @@ class spell_dru_omen_of_clarity : public AuraScript
...
-                else if (spellInfo->SpellFamilyFlags[1] & 0x00000440)
-                {
-                    spellId = SPELL_DRUID_BLESSING_OF_CENARIUS;
-                    chance = 40;
-                }
-                else
-                    return;
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp]: here too

File: src/server/scripts/Spells/spell_druid.cpp [hunk 5]
```diff
@@ -996,1134 +996,916 @@ class spell_dru_omen_of_clarity : public AuraScript
...
-                Unit* target = eventInfo.GetProcTarget();
+        Unit* caster = eventInfo.GetActor();
+        Unit* target = eventInfo.GetProcTarget();
 
-                SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_DRUID_LANGUISH);
-                int32 amount = CalculatePct(static_cast<int32>(damageInfo->GetDamage()), aurEff->GetAmount());
+        SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_DRUID_LANGUISH);
+        int32 amount = CalculatePct(static_cast<int32>(damageInfo->GetDamage()), aurEff->GetAmount());
 
                 ASSERT(spellInfo->GetMaxTicks() > 0);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp]: and here

File: src/server/scripts/Spells/spell_druid.cpp (hunk 2 above)
Comment (Sorikoff) [reply src/server/scripts/Spells/spell_druid.cpp]: That's fault of the IDE. I did auto reformat of the code :\

**Comment by Sorikoff** [comment]
> I think, you should convert complex Spell/Aura scripts too, using RegisterSpellAndAuraScriptPair

Good point! Should I convert in new PR?

**Comment by Shauren** [comment]
Yes, in a new one

**Comment by illfated** [comment]
Your PR #23714 (3 of 3) got merged before this one, so you need to incorporate the changes from that PR to resolve the conflict.

**Comment by Sorikoff** [comment]
Merge conflict must be resolved now.

**Comment by illfated** [comment]
Confirmed, the conflict is resolved.

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**